### ISSUE TITLE
Make it possible to disable ariaHideOutside with ComboBox

### ISSUE THREAD

**Issue by heisand** [adobe/react-spectrum#4937]
### Provide a general summary of the feature here

When navigating with screen readers in the ComboBox, all content around it is hidden for screen readers. This way, the only way one can move to the next element is by using the dismiss button, which can be a bit cumbersome for users.

### 🤔 Expected Behavior?

I asked for an opinion from the authority of universal design of ICT in my country (Norway), and they say:

> It is important to ensure that users can easily navigate out of the list to the next item without having to click a button, as this can improve the user experience. More specifically, there are no WCAG rules that state that it is not OK to be able to navigate to the next element as long as it does not change the context of the page, in which case a warning must be given beforehand or confirmed with a button.

> The availability of the content behind the list is not specified in the documentation as to whether it should be hidden or accessible. However, it is generally recommended to make the content behind the list accessible to screen readers by not using the aria-hidden attribute, this allows users to navigate and interact with the content as needed. Here, then, there is a difference from HTML's specification and React Aria without this necessarily being a breach of the minimum requirements for universal design.

### 😯 Current Behavior

All the content around the ComboBox are hidden by aria-hidden, and screen reader users have to use the dismiss buttons to navigate to the next element.

### 💁 Possible Solution

Add a prop which makes it possible to decide whether to use `ariaHideOutside` or not.

### 🔦 Context

I am trying to implement a user friendly and accessible ComboBox for a large group of users.

### 💻 Examples

Has also been discussed here: 
https://github.com/adobe/react-spectrum/discussions/4871

We discussed that you might could consider allowing users to disable the ariaHideOutside behavior if they aren't portalling the dropdown menu.

It would then be smooth if the ListBox simply closes when the next element is focused. An example of this can be seen in the "Where will you travel" part of https://ruter.no/en.

**Comment by LFDanLu** [comment]
Thanks for filing this and following with the results of your discussion with the ICT. I'll see about bringing this up with the team in our next issue grooming session to see what they think.

**Comment by LFDanLu** [comment]
The team discussed this and are leaning towards exposing a `isNonModal` prop in `useComboBox` similar to the one that already exists in [usePopover](https://github.com/adobe/react-spectrum/blob/145c90329076a996bae53e115bd2488fca6c83a3/packages/%40react-aria/overlays/src/usePopover.ts#L32-L40). If that prop is set, then the we'd turn off ariaHideOutside [here](https://github.com/adobe/react-spectrum/blob/145c90329076a996bae53e115bd2488fca6c83a3/packages/%40react-aria/combobox/src/useComboBox.ts#L300-L304). We'll want to document this new prop as well to make sure people know when and when not to enable it along with the background covered in the [combobox blog post](https://react-spectrum.adobe.com/blog/building-a-combobox.html#portals). This change should only affect the combobox hook, RAC and Spectrum ComboBox shouldn't have that prop exposed since they always render the dropdown in a portal. @heisand would you be interested in contributing this change?

**Comment by heisand** [comment]
@LFDanLu I believe I can look into that 👍

**Comment by LFDanLu** [comment]
Thanks! Feel free to reach out if you have any questions or if you need any pointers on the above.

**Comment by ohazda** [comment]
Hey @heisand, just curious if any progress has been made.

**Comment by heisand** [comment]
> Hey @heisand, just curious if any progress has been made.

Hey, I have simply had a lot going on, so I have not had as much time as wanted.. 

I might be able to work on this towards the end of this month as things might ease down, but I am not sure (I will try) - unless of course, someone else has the capacity to do so.

**Comment by ohazda** [comment]
@LFDanLu Hey, I have a question: What do you think about removing `ariaHideOutside` from this hook completely and making it the responsibility of overlay (popover)?

**Comment by snowystinger** [comment]
@ohazda I'm not sure that would work. In most cases, everything outside the popover is hidden. But in ComboBox, the input is excluded from that. We'd need someway to tell the popover about the input which seems like a mixing of concerns. That said, if you think you've found a way without making a breaking change, we're happy to review it.

**Comment by omarnyte** [comment]
Hello!

I opened an MR for this feature request, although it differs a bit from the original request above. 

**The problem**

Our automated a11y checks are returning a number of `ARIA hidden element must not be focusable or contain focusable elements` violations when the ComboBox popover is open. Based on [this blog post](https://react-aria.adobe.com/blog/building-a-combobox#portals), those `aria-hidden` attributes are critical for screen readers to be able to navigate to the portalled content. 

**The solution**

Apply `inert` (instead of `aria-hidden`) to outside elements when the _modal_ ComboBox popover is open. Continue applying `aria-hidden` to outside elements with _non-modal_ ComboBox popover is open. 

Notably, the [ComboBox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) specifies that "the dialog implements the keyboard interaction defined in the [_modal_ dialog pattern.](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)". But none of [the examples](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/#examples) are actually modal (i.e. they allow interacting with content outside of the popover). The [Date Picker Combobox example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/) sets `aria-modal="true"`, but it still allows interaction with content outside of the popover. 

Please let me know if you need any additional context for a review. Thanks!
